src/lib/cache/cacheManager.ts

```diff
@@ -0,0 +1,356 @@
+    const prefixCommandCategories = await PrefixCommandCategory.find();
+    // Step 2: Get all categories from the cache
+    const cacheKeys = await inMemoryCache.store.keys();
+    // Step 3: Loop over cached categories
+    for (const key of cacheKeys) {
+        if (key.startsWith(`${memoryCachePrefixCategory}:`)) {
+            const categoryName = key.split(':')[1];
+            // Step 3.a: Check if cached category exists in the database list
+            let found = false;
+            for (const dbCategory of prefixCommandCategories) {
+                const { name: dbCategoryName } = dbCategory;
+                if (dbCategoryName.toLowerCase() === categoryName.toLowerCase()) {
+                    found = true;
+                    break;
+                }
+            }
+            // Step 3.b: If not found, remove from cache
+            if (!found) {
+                Logger.debug(`Removing category ${categoryName} from cache`);
+                // eslint-disable-next-line no-await-in-loop
+                await inMemoryCache.del(key);
+            }
+        }
+    }
+    // Step 4: Loop over database categories and update cache
+    for (const dbCategory of prefixCommandCategories) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandCategoryToCache(dbCategory);
+    }
+}
+
+/**
+ * Prefix Command Channel Default Version Cache Management Functions
+ */
+
+export async function clearSinglePrefixCommandChannelDefaultVersionCache(channelDefaultVersion: IPrefixCommandChannelDefaultVersion) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { channelId } = channelDefaultVersion;
+    Logger.debug(`Clearing cache for channel default version for channel "${channelId}"`);
+    await inMemoryCache.del(`${memoryCachePrefixChannelDefaultVersion}:${channelId}`);
+}
+
+export async function loadSinglePrefixCommandChannelDefaultVersionToCache(channelDefaultVersion: IPrefixCommandChannelDefaultVersion) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { channelId, versionId } = channelDefaultVersion;
+    const version = await PrefixCommandVersion.findById(versionId);
+    if (version) {
+        Logger.debug(`Loading default version for channel ${channelId} to cache`);
+        await inMemoryCache.set(`${memoryCachePrefixChannelDefaultVersion}:${channelId}`, version.toObject());
+    }
+}
+
+export async function loadAllPrefixCommandChannelDefaultVersionsToCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    const PrefixCommandChannelDefaultVersions = await PrefixCommandChannelDefaultVersion.find();
+
+    for (const defaultVersion of PrefixCommandChannelDefaultVersions) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandChannelDefaultVersionToCache(defaultVersion);
+    }
```

Comment:
```suggestion
    await Promise.all(
        PrefixCommandChannelDefaultVersions
            .map((defaultVersion) => loadSinglePrefixCommandChannelDefaultVersionToCache(defaultVersion)),
    );
```
Can't wait till prettier does the formtting for us...